fix(indy-validator): fix package dependencies

outSH opened the pull request:
- Update indy validator python dependencies.
- Add README chapter on updating python dependencies to simplify this process in the future.
- Use pinend ubuntu base image in indy-sdk-cli dockerfile.
- Do some minor README improvements and cleanups.

Tested with `discounted-asset-trade` (should work without an issue now, at least dockerless one)

Depends on: #2596

Signed-off-by: Michal Bajer <michal.bajer@fujitsu.com>

izuru0 reviewed (approved):
LGTM

petermetz reviewed (changes requested):
@outSH Thank you and LGTM, but please rebase onto upstream/main and resolve the merge conflicts!

outSH commented:
> @outSH Thank you and LGTM, but please rebase onto upstream/main and resolve the merge conflicts!

Done

petermetz commented on tools/docker/fabric-all-in-one/Dockerfile_v1.4.x:
```diff
@@ -41,7 +41,7 @@ RUN apk add --no-cache libc6-compat
 RUN mkdir -p $CACTUS_CFG_PATH
 # OpenSSH - need to have it so we can shell in and install/instantiate contracts
-RUN apk add --no-cache openssh augeas
+RUN apk add --no-cache augeas
```
@outSH I think we need the OpenSSH server installed for contract deployments. Is this somehow part of the indy-validator fix or just conflict resolution gone wrong?

LGTM otherwise, but I want to make sure about this one.

outSH replied:
@petermetz Yes, sorry, this is leftover from temporary change from dependent commit when I was testing this change. This has been reverted, openssh remains.

petermetz reviewed (changes requested):
@outSH I just spotted that the parent PR's commit is still in here with a few lines of diff, but the parent PR has already been merged onto main so please squash or remove the second commit otherwise when we merge this there'll be two commits with the same message but one of them will have the actual diff while the other just a couple of these lines of changes.

outSH commented:
> @outSH I just spotted that the parent PR's commit is still in here with a few lines of diff, but the parent PR has already been merged onto main so please squash or remove the second commit otherwise when we merge this there'll be two commits with the same message but one of them will have the actual diff while the other just a couple of these lines of changes.

@petermetz You're right, sorry for that. Fixed, rebased and ready for merge

petermetz reviewed (approved):
> @petermetz You're right, sorry for that. Fixed, rebased and ready for merge

@outSH No worries, thank you very much for the updates! LGTM
